### DISCUSSION TITLE
MTL files with missing information have default values

### DISCUSSION THREAD

**Discussion by chickenSandwich** [assimp/assimp#3740] [Q&A] [answered]
Greetings,
  When I load a simple OBJ/MTL file,  I noticed that the behavior of aiGetMaterialFloatArray is to return AI_SUCCESS even if the MTL file does not have a value for certain properties (such as Ns which refers to shininess).  According to the Material structure (https://github.com/assimp/assimp/blob/master/code/AssetLib/Obj/ObjFileData.h#L179), sensible defaults are set.
  I'm wondering how to detect that a material is missing a certain property so that I can assume my own defaults (without opening and parsing the file of course).  Additionally, I was wondering if these defaults are consistent with other model types.  I poked around the AssetLib folders a bit and observed similar defaults.  As I am new to 3D modelling, I am trying to develop a protocol for handling materials.  I welcome any advice.
Thank-you.

**Comment by kimkulling** [comment, marked as answer]
Normally aiMaterial::get() will return null if the value is not set in the material.
